### PR TITLE
Fix build failure caused by large test failure responses

### LINKED ISSUE #451: JMeter Responses are too large for database
**Description:**
During the Jmeter execution, we save the requests and responses in the database. however, during the selenium tests, these responses are too large to be saved (  HTML pages ). We need to handle this issue in a suitable manner.
For more info [1]
[1] - https://testgrid-live-dev.private.wso2.com/job/wso2is-5.4.1-LTS/16/consoleFull

**Suggested Labels:**
Severity/Major
Priority / High
Type/Bug

**Affected Product Version:**
v0.9.0-m-11

### PR DESCRIPTION
**Pull request by azinneera**
## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: distribution/bin/log4j2.xml [hunk 1]
```diff
@@ -67,5 +67,9 @@
         <Logger name="org.eclipse.persistence" level="info" additivity="false">
             <AppenderRef ref="Routing"/>
         </Logger>
+        <Logger name="org.wso2.testgrid.automation.executor.JMeterResultCollector" additivity="false">
+            <AppenderRef ref="Routing" level="info"/>
+            <AppenderRef ref="TESTGRID_CONSOLE" level="error" />
+        </Logger>
```
Comment (kasunbg) [inline distribution/bin/log4j2.xml]: Logging into Routing and TESTGRID_CONSOLE is the default behavior (as defined by 'Root' above). Hence, we don't need this.

File: automation/src/main/java/org/wso2/testgrid/automation/executor/JMeterResultCollector.java [hunk 2]
```diff
@@ -60,11 +63,11 @@ public void sampleOccurred(SampleEvent sampleEvent) {
                                             "\", \"Status code\": \"",
                                             result.getResponseCode().replaceAll("\"", "\\\""),
                                             "\", \"Response Message\": \"",
-                                            result.getResponseMessage().replaceAll("\"", "\\\""),
-                                            "\", \"Sampler Data\": \"",
-                                            result.getSamplerData().replaceAll("\"", "\\\""), "\"}");
+                                            result.getResponseMessage().replaceAll("\"", "\\\""));
```
Comment (kasunbg) [inline automation/src/main/java/org/wso2/testgrid/automation/executor/JMeterResultCollector.java]: Can we show a portion of the sampler data? It is useful when the response sizes r useful. I'm thinking of something like following?

String samplerData = result.getSamplerData();
if (samplerData.size() < 5000) {
  samplerData = samplerData.substring(0, 5000) + "\n log truncated..."
}

File: automation/src/main/java/org/wso2/testgrid/automation/executor/JMeterResultCollector.java [hunk 3]
```diff
@@ -60,11 +63,11 @@ public void sampleOccurred(SampleEvent sampleEvent) {
...
             // Persist result to the database
             testCaseUOW.persistTestCase(result.getSampleLabel(), testScenario, result.isSuccessful(), failureMessage);
+            logger.info(StringUtil.concatStrings(failureMessage, "\", \"Sampler Data\": \"",
+                    result.getSamplerData().replaceAll("\"", "\\\""), "\"}"));
```
Comment (kasunbg) [inline automation/src/main/java/org/wso2/testgrid/automation/executor/JMeterResultCollector.java]: I think we need to print sampleLabel, testScenario along with this. Basically, we need more context.. :)

Comment (azinneera) [reply automation/src/main/java/org/wso2/testgrid/automation/executor/JMeterResultCollector.java]: Fixed in 1a609df

File: automation/src/main/java/org/wso2/testgrid/automation/executor/JMeterResultCollector.java (hunk 2 above)
Comment (azinneera) [reply automation/src/main/java/org/wso2/testgrid/automation/executor/JMeterResultCollector.java]: Fixed in 1a609df

File: distribution/bin/log4j2.xml (hunk 1 above)
Comment (azinneera) [reply distribution/bin/log4j2.xml]: Fixed by introducing a separate logger for appending jmeter test results in 50ccef6

File: automation/src/main/java/org/wso2/testgrid/automation/executor/JMeterResultCollector.java [hunk 4]
```diff
@@ -33,6 +35,8 @@
  */
 public class JMeterResultCollector extends ResultCollector {
 
+    private static final Logger jmeterLogger = LoggerFactory.getLogger(JMeterResultCollector.class.getName()
+            + ".JMeterLogger");
```
Comment (kasunbg) [inline automation/src/main/java/org/wso2/testgrid/automation/executor/JMeterResultCollector.java]: *JMeterResultLogger?

File: automation/src/main/java/org/wso2/testgrid/automation/executor/JMeterResultCollector.java [hunk 5]
```diff
@@ -54,6 +59,11 @@ public void sampleOccurred(SampleEvent sampleEvent) {
             TestCaseUOW testCaseUOW = new TestCaseUOW();
             super.sampleOccurred(sampleEvent);
             SampleResult result = sampleEvent.getResult();
+            String samplerData = result.getSamplerData();
+
+            if (samplerData.length() > 5000) {
```
Comment (sameerawickramasekara) [inline automation/src/main/java/org/wso2/testgrid/automation/executor/JMeterResultCollector.java]: Value is hardcoded here, should use the variable `FAILURE_MESSAGE_LIMIT` defined above

Comment (azinneera) [reply automation/src/main/java/org/wso2/testgrid/automation/executor/JMeterResultCollector.java]: Fixed in 6d26744

**Pull request merged**